### PR TITLE
[Snackbar] Update content padding for new snackbar, leave old snackbar padding as is.

### PR DESCRIPTION
**Pull request by yarneo**
Legacy snackbar should visually look the same before and after changes.

**Before pics for legacy and non-legacy snackbar:**
iPhone X, old snackbar, short text, button not tapped
screen shot 2018-03-29 at 4 35 41 pm

iPhone X, old snackbar, long text, button not tapped
screen shot 2018-03-29 at 4 35 45 pm

iPhone X, new snackbar, short text, button not tapped
screen shot 2018-03-29 at 4 35 50 pm

iPhone X, new snackbar, long text, button not tapped
screen shot 2018-03-29 at 4 35 54 pm

iPhone X, new snackbar, short text, button tapped
untitled

iPhone X, old snackbar, short text, button tapped
untitled2

**After pics for legacy and non-legacy snackbar:**
iPhone X, new snackbar, long text, button not tapped
screen shot 2018-03-29 at 4 26 49 pm

iPhone X, new snackbar, short text, button not tapped
screen shot 2018-03-29 at 4 27 05 pm

iPhone X, old snackbar, short text, button not tapped
screen shot 2018-03-29 at 4 27 15 pm

iPhone X, old snackbar, long text, button not tapped
screen shot 2018-03-29 at 4 27 20 pm

iPhone X, new snackbar, long text, button tapped
untitled

iPhone X, new snackbar, short text, button tapped
untitled2

iPhone X, old snackbar, long text, button tapped
untitled3

iPhone X, old snackbar, short text, button tapped
untitled4

iPhone 8, new snackbar, long text, button not tapped
screen shot 2018-03-29 at 4 29 11 pm

iPhone 8, old snackbar, long text, button not tapped
screen shot 2018-03-29 at 4 29 21 pm

### CODE REVIEW

File: components/Snackbar/src/MDCSnackbarMessageView.m [hunk 1]
```diff
@@ -613,6 +624,10 @@ - (void)updateButtonFont {
   [self setNeedsLayout];
 }
 
+- (CGFloat)buttonContentPadding {
+  return MDCSnackbarMessage.usesLegacySnackbar ? kLegacyButtonPadding : kButtonPadding;
```
Comment (ianegordon) [inline components/Snackbar/src/MDCSnackbarMessageView.m]: This is a one line method that is only called from two places, consider removing it.
If you do want to keep it, make it a class method.  (Generally, if you don't reference self in a method, it should be a class method)

File: components/Snackbar/src/MDCSnackbarMessageView.m [hunk 2]
```diff
@@ -833,7 +859,7 @@ - (NSArray *)horizontalButtonLayoutConstraints {
     @"kTitleImagePadding" : @(kTitleImagePadding),
     @"kBorderMargin" : @(kBorderWidth),
     @"kTitleButtonPadding" : @(kTitleButtonPadding),
-    @"kButtonPadding" : @(kButtonPadding),
+    @"kButtonPadding" : @([self buttonContentPadding]),
```
Comment (ianegordon) [inline components/Snackbar/src/MDCSnackbarMessageView.m]: :/ This naming is not great.  Historically 'k' has prefixed static const values.  Out of scope for this PR, but not great.

File: components/Snackbar/src/MDCSnackbarMessageView.m [hunk 3]
```diff
@@ -940,7 +966,8 @@ - (CGFloat)contentSafeBottomInset {
 }
 
 - (UIEdgeInsets)safeContentMargin {
-  UIEdgeInsets contentMargin = kContentMargin;
+  UIEdgeInsets contentMargin =
+      MDCSnackbarMessage.usesLegacySnackbar ? kLegacyContentMargin : kContentMargin;
```
Comment (ianegordon) [inline components/Snackbar/src/MDCSnackbarMessageView.m:968]: I would mimic this pattern rather than a creating a method for buttonContentPadding.

**Review by ianegordon — DISMISSED** [review]

File: components/Snackbar/src/MDCSnackbarMessageView.m (hunk 3 above)
Comment (yarneo) [reply components/Snackbar/src/MDCSnackbarMessageView.m:968]: done

**Review by ianegordon — APPROVED** [review]

**Pull request merged**